src/Birkhoff_polytope.jl

```diff
@@ -0,0 +1,453 @@
+                push!(fixed_to_one_cols, col_orig)
+
+                push!(delete_index_map_rows, i)
+                push!(delete_index_map_cols, j)
+            end
+        end
+    end
+
+    unique!(delete_index_map_rows)
+    unique!(delete_index_map_cols)
+    sort!(delete_index_map_rows)
+    sort!(delete_index_map_cols)
+    deleteat!(index_map_rows, delete_index_map_rows)
+    deleteat!(index_map_cols, delete_index_map_cols)
+
+    nreduced = length(index_map_rows)
+    type = typeof(direction[1, 1])
+    d2 = ones(Union{type,Missing}, nreduced, nreduced)
+    for j = 1:nreduced
+        for i = 1:nreduced
+            row_orig = index_map_rows[i]
+            col_orig = index_map_cols[j]
+            if blmo.append_by_column
+                orig_linear_idx = (col_orig - 1) * n + row_orig
+            else
+                orig_linear_idx = (row_orig - 1) * n + col_orig
+            end
+            # interdict arc when fixed to zero
+            if ub[orig_linear_idx] <= eps() || x[row_orig, col_orig] <= eps()
+                if blmo.append_by_column
+                    d2[i, j] = missing
+                else
+                    d2[j, i] = missing
+                end
+            else
+                if blmo.append_by_column
+                    d2[i, j] = direction[row_orig, col_orig]
+                else
+                    d2[j, i] = direction[col_orig, row_orig]
+                end
+            end
+        end
+    end
+
+    m = SparseArrays.spzeros(n, n)
+    for (i, j) in zip(fixed_to_one_rows, fixed_to_one_cols)
+        m[i, j] = 1
+    end
+
+    res_mat = Hungarian.munkres(d2)
+    (rows, cols, vals) = SparseArrays.findnz(res_mat)
+    @inbounds for i in eachindex(cols)
+        m[index_map_rows[rows[i]], index_map_cols[cols[i]]] = (vals[i] == 2)
+    end
+
+    m = if blmo.append_by_column
+        # Convert sparse matrix to sparse vector by columns
+        I, J, V = SparseArrays.findnz(m)
+        linear_indices = (J .- 1) .* n .+ I
+        SparseArrays.sparsevec(linear_indices, V, n^2)
+    else
+        # Convert sparse matrix to sparse vector by rows (transpose first)
+        mt = SparseArrays.sparse(LinearAlgebra.transpose(m))
+        I, J, V = SparseArrays.findnz(mt)
+        linear_indices = (J .- 1) .* n .+ I
+        SparseArrays.sparsevec(linear_indices, V, n^2)
+    end
+
+    return m
+end
+
+"""
+LMO-like operation which computes a vertex minimizing in `direction` on the face defined by the current fixings.
+Fixings are maintained by the oracle (or deduced from `x` itself).
+"""
+function Boscia.dicg_maximum_step(blmo::BirkhoffBLMO, direction, x; kwargs...)
+    n = blmo.dim
+
+    direction =
+        blmo.append_by_column ? reshape(direction, (n, n)) :
+        transpose(reshape(direction, (n, n)))
+    x = blmo.append_by_column ? reshape(x, (n, n)) : transpose(reshape(x, (n, n)))
+    return FrankWolfe.dicg_maximum_step(FrankWolfe.BirkhoffPolytopeLMO(), direction, x)
+end
```

Comment:
To avoid duplicate code, it would be cool to build this BLMO in a way such that it can also be used in Frank-Wolfe, I.e. we can also remove the BirkhoffLMO from FrankWolfe.jl.